# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Selected Lane:** Lane 2 — Refresh / Content Opportunity Scoring

I choose this lane because content decay is one of the most expensive and silent problems in search intelligence. Based on the operational feedback and insights shared by Mr. Mirza, relying purely on paid ads provides only a temporary traffic spike, and visitor engagement halts completely the moment the advertisement budget stops, creating unsustainable marketing expenses. By utilizing machine learning to mathematically predict and prioritize which existing high-value assets are genuinely declining, we can organically refresh them. This strategy builds permanent, cost-free search visibility, directly reducing ad dependency and optimizing the client's long-term editorial budget.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

* **The Search Question:** Which specific URLs are experiencing a true performance decline, and how should content engineering teams prioritize them for an organic refresh to prevent further organic traffic loss?
* **Unit of Analysis:** A single URL (page) level.
* **The Decision & Action:** This analysis directly improves the editorial budgeting and resource allocation decision. The action taken is generating an automated, prioritized weekly review queue for content writers, telling them exactly which top-K pages to update next to build sustainable organic growth.
* **The Cost of a Wrong Call:**
  * *False Positive (Flagging a healthy page as decaying):* Causes the company to waste hours of human writing budget and money rewriting a page that didn't need it, incurring a high opportunity cost.
  * *False Negative (Missing a genuinely decaying page):* Means a valuable traffic asset drops to zero. As highlighted by Mr. Mirza, this forces the business to rely heavily on paid advertisements to recover the lost traffic. Since ad engagement halts completely the moment the budget stops, this mistake results in unsustainable, continuous marketing expenses and high customer acquisition costs.
* **Why Data/ML Helps:** Human intuition cannot track thousands of URLs simultaneously across shifting search patterns. A readable machine learning model separates daily statistical noise from systemic downward trends, prioritizing items by exposure, volume, and financial risk.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [16]:
import pandas as pd
import os

file_path = "content_refresh_anonymized.csv"

if os.path.exists(file_path):
    df = pd.read_csv(file_path)
    print("✓ CSV successfully loaded and defined as 'df'!")
    print(f"The dataset contains {df.shape[0]} rows and {df.shape[1]} columns.")
else:
    print(f"[ERROR] '{file_path}' not found!")
    print("Please make sure you have uploaded the file into the 'data/raw/' directory via the Colab sidebar.")

✓ CSV successfully loaded and defined as 'df'!
The dataset contains 30000 rows and 44 columns.


In [17]:
df.head()

,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [18]:
df.shape

(30000, 44)

In [19]:
df["trend_direction"].value_counts()

,count
trend_direction,
down,16262
stable,5962
up,4388
new,2236
flat,1152


In [20]:
base_decay_rate = df['trend_direction'].str.lower().eq('down').mean()

In [21]:
stale_pages = df[df['days_since_last_update'] >= 180]
stale_decay_rate = stale_pages['trend_direction'].str.lower().eq('down').mean()

# Print the outputs to see the real numbers
print(f"Total Unique Pages: {total_pages}")
print(f"Overall Decaying Rate (Base Rate): {base_decay_rate * 100:.1f}%")
print(f"Declining Rate for Stale Pages (>= 180 days): {stale_decay_rate * 100:.1f}%")

Total Unique Pages: 30000
Overall Decaying Rate (Base Rate): 54.2%
Declining Rate for Stale Pages (>= 180 days): 47.1%


### Data Insights & Business Justification

Based on the initial data execution, we observed the following metrics from our inventory:
* **Total Unique Pages Analysed:** 30,000 URLs
* **Overall Decaying Rate (Base Rate):** 54.2%
* **Declining Rate for Stale Pages (>= 180 days):** 47.1%

**Honest Interpretation:** The data reveals a critical systemic issue: more than half of the entire page inventory (54.2%) is directionally declining in performance. While stale pages show a slightly lower decay rate of 47.1%, the sheer volume means thousands of old assets are consistently losing organic visibility.

This mathematically backs up the operational risk shared by Mr. Mirza: when 54.2% of your site is decay-exposed, masking this loss with temporary paid advertisements creates an unsustainable cost structure. We cannot solve a 30,000-page operational problem with manual oversight or ad budgets. Building an ML scoring engine to isolate and prioritize the high-impact candidates within this decaying pool is definitively worth our next 7 weeks of development.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

* **What I can claim:** This model serves strictly as an operational decision-support system. It can isolate historical performance trends, identify observed correlation patterns between content age and traffic drop, and directionally rank which existing URLs present the highest exposure risk. This allows the editorial team to optimize their limited refresh budget.
* **What I can NEVER claim:** This work does not prove definitive causality. It cannot predict sudden, unmeasured Google Core Algorithm updates, nor can it guarantee that executing a content refresh will automatically or instantly restore a page's peak historical ranking position. We predict directional risk based on historical data, not absolute search engine certainty.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.